redaxo/src/core/lib/console/setup/run.php

```diff
@@ -0,0 +1,348 @@
+            ->addOption('--db-setup', null, InputOption::VALUE_REQUIRED, 'Database setup mode e.g. "normal", "override" or "import"')
+            ->addOption('--db-import', null, InputOption::VALUE_REQUIRED, 'Database import filename if "import" is used as --db-setup')
+            ->addOption('--admin-username', null, InputOption::VALUE_REQUIRED, 'Creates a redaxo admin user with the given username')
+            ->addOption('--admin-password', null, InputOption::VALUE_REQUIRED, 'Sets the password for the admin user account')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = $this->getStyle($input, $output);
+
+        $configFile = rex_path::coreData('config.yml');
+        $config = array_merge(
+            rex_file::getConfig(rex_path::core('default.config.yml')),
+            rex_file::getConfig($configFile)
+        );
+        $config['setup'] = true;
+
+        $requiredValue = static function ($value) {
+            if (empty($value)) {
+                throw new InvalidArgumentException('Value required');
+            }
+            return $value;
+        };
+
+        // ---------------------------------- Step 1 . Language
+        $io->title('Step 1 of 6 / Language');
+        $langs = [];
+        foreach (rex_i18n::getLocales() as $locale) {
+            $langs[$locale] = rex_i18n::msgInLocale('lang', $locale);
+        }
+        ksort($langs);
+
+        if (null === $input->getOption('lang')) {
+            $config['lang'] = $io->askQuestion(new ChoiceQuestion('Please select a language', $langs));
+        } else {
+            $lang = $input->getOption('lang');
+            if (!$lang || in_array($lang, $langs, true)) {
+                throw new InvalidArgumentException('Unknown lang "' . $lang . '" specified');
+            }
+            $config['lang'] = $lang;
+            $io->success('Language "'.$lang.'" selected.');
+        }
+
+        // ---------------------------------- Step 2 . license
+        $io->title('Step 2 of 6 / Licence');
+
+        if (false === $input->getOption('agree-licence')) {
+            $license_file = rex_path::base('LICENSE.md');
+            $license = rex_file::get($license_file);
+            $io->writeln($license);
+            if (!$io->confirm('Accept licence terms and conditions?')) {
+                $io->error('You need to accept licence terms and conditions');
+                return 1;
+            }
+        } else {
+            if (null === $input->getOption('agree-licence')) {
+                $io->error('You need to accept licence terms and conditions');
+                return 1;
+            }
+            $io->success('You accepted licence terms and conditions');
+        }
+
+        // ---------------------------------- Step 3 . Perms, Environment
+        $io->title('Step 3 of 6 / System check');
+
+        // Embed existing check
+        $command = $this->getApplication()->find('setup:check');
```

Comment:
Wenn ich das Setup interaktiv durchlaufe, komme ich bei Schritt 3 nicht weiter. Dort wird `setup:check` aufgerufen, und das bricht ab, wenn noch keine `config.yml` besteht:

![Screenshot 2020-01-13 at 21 49 56](https://user-images.githubusercontent.com/1297466/72291040-cc3b2a80-364e-11ea-8537-24f30a20b978.png)

Reply:
okay in `setup:check` erwartet der schon eine `config.yml`....

Der Command sollte IMO eigentlich nur wie im GUI Setup nur die Umgebung (php version, permissions, etc.) prüfen und eigentlich nicht voraussetzen, dass schon eine `config.yml` vorhanden ist.

Reply:
@gharlan, @staabm meinung?

Reply:
wobei ohne config.yml dort auch nicht die db geprüft werden kann, oder?

Hier im setup-run-command hingegen ist man ja noch gar nicht bei der db.

Ggf passt es doch nicht, den command so direkt hier zu verwenden?

Reply:
> wobei ohne config.yml dort auch nicht die db geprüft werden kann, oder?

in `setup:check` korrekt, hier gehts auch ohne.

Ich vergleichs halt mit der GUI Variante und hätte eigentlich vermutet, dass der `setup:check` command equivalent ist.
![image](https://user-images.githubusercontent.com/21292755/72295621-3c9a7980-3658-11ea-98ea-7c45e3e798b6.png)

Aber ja, sonst würde ich die checks eher in `rex_setup` verlagern und von beiden nur ein call machen.

Reply:
> Ich vergleichs halt mit der GUI Variante und hätte eigentlich vermutet, dass der `setup:check` command equivalent ist.

Würde ich glaube ich nicht ganz so sehen. Also der setup:check-Command ist für mich schon eher ein übergreifender Command für die Frage "Passt (noch) alles mit dem System?", und somit mehr als der Check-Schritt während des Setups. Eben noch plus DB-Check.

Reply:
habs jetzt aus `setup:check` gecloned.